src/test/java/seedu/address/storage/JsonAdaptedTuitionClassTest.java

```diff
@@ -0,0 +1,110 @@
+package seedu.address.storage;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static seedu.address.storage.JsonAdaptedTuitionClass.MISSING_FIELD_MESSAGE_FORMAT;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalTimestable.IB_PHYSICS;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.tuitionclass.ClassName;
+import seedu.address.model.tuitionclass.ClassTiming;
+import seedu.address.model.tuitionclass.Location;
+import seedu.address.model.tuitionclass.Rate;
+
+public class JsonAdaptedTuitionClassTest {
+    private static final String INVALID_CLASS_NAME = " ";
+    private static final String INVALID_CLASS_TIMING = "MON 11:00-10:00";
+    private static final String INVALID_RATE = "Sixty";
+    private static final String INVALID_LOCATION = " ";
+    private static final List<String> INVALID_STUDENT_NAME_LIST = new ArrayList<>(Arrays.asList(" "));
+
+    private static final String VALID_CLASS_NAME = IB_PHYSICS.getClassName().toString();
+    private static final String VALID_CLASS_TIMING = IB_PHYSICS.getClassTiming().toString();
+    private static final String VALID_RATE = IB_PHYSICS.getRate().toString();
+    private static final String VALID_LOCATION = IB_PHYSICS.getLocation().toString();
+    private static final List<String> VALID_EMPTY_STUDENT_NAME_LIST = new ArrayList<>();
+
+    @Test
+    public void toModelType_validTuitionClassDetails_returnsPerson() throws Exception {
+        JsonAdaptedTuitionClass tuitionClass = new JsonAdaptedTuitionClass(IB_PHYSICS);
+        assertEquals(IB_PHYSICS, tuitionClass.toModelType());
+    }
+
+    @Test
+    public void toModelType_nullStudentList_success() throws Exception {
```

Comment:
Should the method name be toModelType_emptyStudentList_success instead? Should there be another test where the studentList is the value `null` too?